File: backend/app/schemas/planner.py
from pydantic import BaseModel
from datetime import date
from typing import List

class PlannerGenerateRequest(BaseModel):
    course_name:str
    exam_date: date
    topics: List[str]
    daily_study_hours: float
    textbook: str | None = None

class StudyTaskResponse(BaseModel):
    id: int
    title: str
    duration_minutes: int
    task_type: str
    position: int

class DailyPlanResponse(BaseModel):
    id: int
    day: date
    tasks: List[StudyTaskResponse]

class UnscheduledTaskResponse(BaseModel):
    title: str
    duration_minutes: int
    task_type: str
    topic: str

class PlannerGenerateResponse(BaseModel):
    course_id: int
    course_name: str
    exam_date: date
    daily_plans: List[DailyPlanResponse]
    unscheduled: List[UnscheduledTaskResponse] = []
    warning: str | None = None

class PlannerCourseResponse(PlannerGenerateResponse):
    pass

File: frontend/src/types/planner.ts
export type PlannerStudyTask = {
  id: number;
  title: string;
  duration_minutes: number;
  task_type: string;
  position: number;
};

export type PlannerDailyPlan = {
  id: number;
  day: string;
  tasks: PlannerStudyTask[];
};

export type PlannerUnscheduledTask = {
  title: string;
  duration_minutes: number;
  task_type: string;
  topic: string;
};

export type PlannerGenerateRequest = {
  course_name: string;
  exam_date: string;
  topics: string[];
  daily_study_hours: number;
  textbook?: string;
};

export type PlannerGenerateResponse = {
  course_id: number;
  course_name: string;
  exam_date: string;
  daily_plans: PlannerDailyPlan[];
  unscheduled?: PlannerUnscheduledTask[];
  warning?: string | null;
};
